ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -271,36 +271,101 @@ private void addNewDataNodes() {
   }
 
   /**
-   * Goes through each existing partition and changes the {@link PartitionState} for the replicas in each of the
-   * instances that hosts a replica for this partition, if it has changed and is different from the current state
-   * in Helix.
-   * @param partition the partition whose {@link PartitionState} may have to be updated.
+   * Goes through each existing partition and updates the {@link PartitionState} and capacity information for the
+   * replicas in each of the instances that hosts a replica for this partition, if it has changed and is different from
+   * the current information in the static cluster map.
+   * @param partition the partition whose {@link PartitionState} and/or capacity may have to be updated.
    */
-  private void updatePartitionStateIfChanged(Partition partition) {
+  private void updatePartitionInfoIfChanged(Partition partition) {
     for (Map.Entry<String, HelixAdmin> entry : adminForDc.entrySet()) {
       String dcName = entry.getKey();
       HelixAdmin dcAdmin = entry.getValue();
       String partitionName = Long.toString(partition.getId());
+      long replicaCapacityInStatic = partition.getReplicaCapacityInBytes();
       boolean isSealed = partition.getPartitionState().equals(PartitionState.READ_ONLY);
       List<ReplicaId> replicaList = getReplicasInDc(partition, dcName);
       for (ReplicaId replicaId : replicaList) {
         DataNodeId node = replicaId.getDataNodeId();
         String instanceName = getInstanceName(node);
         InstanceConfig instanceConfig = dcAdmin.getInstanceConfig(clusterName, instanceName);
-        List<String> currentSealedPartitions = instanceConfig.getRecord().getListField(ClusterMapUtils.SEALED_STR);
-        List<String> newSealedPartitionsList = new ArrayList<>(currentSealedPartitions);
-        if (isSealed && !currentSealedPartitions.contains(partitionName)) {
-          newSealedPartitionsList.add(partitionName);
-        } else if (!isSealed && currentSealedPartitions.contains(partitionName)) {
-          newSealedPartitionsList.remove(partitionName);
+        boolean shouldSetInstanceConfig = false;
```

Comment:
I actually prefer it the way it is, as we are setting the instanceConfig later. The "update" prefix is generally being used for method names in this class.